### PR TITLE
Better approach to register plugins

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Better approach to register plugins

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/plugins_registry.py [hunk 1]
```diff
@@ -134,13 +133,8 @@ def is_register_plugins_overridden(plugin: Callable) -> bool:
...
-            module = importlib.import_module(".".join([base_module, module]))
-            for _, mod in getmembers(module, isclass):
-                if issubclass(mod, TrainingTypePlugin) and is_register_plugins_overridden(mod):
-                    mod.register_plugins(TrainingTypePluginsRegistry)
-                    break
+    module = importlib.import_module(base_module)
+    for _, mod in getmembers(module, isclass):
+        if issubclass(mod, TrainingTypePlugin) and is_register_plugins_overridden(mod):
+            mod.register_plugins(TrainingTypePluginsRegistry)
+            break
```
Comment (ananthsub) [inline pytorch_lightning/plugins/plugins_registry.py:140]: Why the break?

**Review by ananthsub — COMMENTED** [review]
Eventually would we enable plugins outside this repo to add to this registry?

File: pytorch_lightning/plugins/plugins_registry.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/plugins/plugins_registry.py]: Done!

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 2]
```diff
@@ -332,3 +332,12 @@ def predict_step(self, *args, **kwargs):
     def post_training_step(self):
         if not self.lightning_module.automatic_optimization:
             self.model.require_backward_grad_sync = True
+
+    @classmethod
```
Comment (ananthsub) [inline pytorch_lightning/plugins/training_type/ddp.py:336]: I think we should split this to a separate PR to ensure this PR does only one thing

**Review by SeanNaren — APPROVED** [review]
I'll approve but as ananth said, let's save the other config for a separate PR! 

I have a few concerns of the name we gave ddp with find_unused_params=false since its quite long and I'd like the alias to be as small as possible :)

**Comment by kaushikb11** [comment]
I have a few concerns of the name we gave ddp with find_unused_params=false since its quite long and I'd like the alias to be as small as possible :)

Haha, @SeanNaren. I wasn't the biggest fan of the name either lol. Suggestions are welcome :))

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]
lgtm
